src/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -364,61 +362,14 @@ public SAMRecordIterator assertSorted(final SortOrder sortOrder) {
     @Override
     public CloseableIterator<SAMRecord> queryAlignmentStart(final String sequence,
                                                             final int start) {
-        long[] filePointers = null;
-
-        // Hit the index to determine the chunk boundaries for the required data.
         final SAMFileHeader fileHeader = getFileHeader();
         final int referenceIndex = fileHeader.getSequenceIndex(sequence);
-        if (referenceIndex != -1) {
-            final BAMIndex fileIndex = getIndex();
-            final BAMFileSpan fileSpan = fileIndex.getSpanOverlapping(
-                    referenceIndex, start, -1);
-            filePointers = fileSpan != null ? fileSpan.toCoordinateArray()
-                    : null;
-        }
-
-        if (filePointers == null || filePointers.length == 0)
-            return emptyIterator;
-
-        Container container;
-        final SeekableStream seekableStream = getSeekableStreamOrFailWithRTE();
-        for (int i = 0; i < filePointers.length; i += 2) {
-            final long containerOffset = filePointers[i] >>> 16;
-
-            try {
-                seekableStream.seek(containerOffset);
-                iterator.nextContainer();
-
-                if (iterator.jumpWithinContainerToPos(fileHeader.getSequenceIndex(sequence), start)) {
-                    return new IntervalIterator(iterator, new QueryInterval(referenceIndex, start, -1));
-                }
-            } catch (final IOException e) {
-                throw new RuntimeIOException(e);
-            } catch (IllegalAccessException e) {
-                throw new SAMException(e);
-            }
-        }
-        throw new SAMException("Failed to query alignment start: " + sequence + " at " + start);
+        return new CRAMIntervalIterator(new QueryInterval[]{new QueryInterval(referenceIndex, start, -1)}, true);
     }
 
     CloseableIterator<SAMRecord> query(final int referenceIndex,
                                        final int start, final int end, final boolean overlap) throws IOException {
```

Comment:
Note to self: Delete this method. I meant to remove it as its no longer used internally and should just be deleted (its not part of the public query interface, and is confusing since it reverses the polarity of overlap).